### PR TITLE
APPS-806 optional fix

### PR DESCRIPTION
**Pull request by mhrvol**

### CODE REVIEW

File: core/src/main/scala/dx/core/ir/ValueSerde.scala [hunk 1]
```diff
@@ -430,10 +429,9 @@ object ValueSerde extends DefaultJsonProtocol {
           // in fieldTypes that do not appear in fields are optional
           val keys1 = fields.keySet
           val keys2 = fieldTypes.keySet
-          val extra = keys2.diff(keys1)
-          if (extra.nonEmpty) {
+          if (!keys1.subsetOf(keys2)) {
             throw ValueSerdeException(
-                s"struct ${name} value has members that do not appear in the struct definition: ${extra}"
+                s"struct definition: ${keys1}, has members that do not appear in struct ${name}"
```
Comment (jdidion) [inline core/src/main/scala/dx/core/ir/ValueSerde.scala]: This message is wrong. `keys1` are the keys in the struct value.

File: core/src/main/scala/dx/core/languages/cwl/CwlUtils.scala [hunk 2]
```diff
@@ -398,10 +398,9 @@ object CwlUtils {
           // in memberTypes that do not appear in members are optional
           val keys1 = fields.keySet
           val keys2 = record.fields.keySet
-          val extra = keys2.diff(keys1)
-          if (extra.nonEmpty) {
+          if (!keys1.subsetOf(keys2)) {
             throw new Exception(
-                s"struct ${record.name} value has members that do not appear in the struct definition: ${extra}"
+                s"struct definition: ${keys1}, has members that do not appear in struct ${record.name}"
```
Comment (jdidion) [inline core/src/main/scala/dx/core/languages/cwl/CwlUtils.scala]: same problem here. Also, they're records in CWL, not structs (that was my mistake).

File: core/src/main/scala/dx/core/languages/wdl/WdlUtils.scala [hunk 3]
```diff
@@ -488,10 +488,9 @@ object WdlUtils {
     // in memberTypes that do not appear in members are optional
     val keys1 = fieldValues.keySet
     val keys2 = fieldTypes.keySet
-    val extra = keys2.diff(keys1)
-    if (extra.nonEmpty) {
+    if (!keys1.subsetOf(keys2)) {
       throw new Exception(
-          s"struct ${name} value has members that do not appear in the struct definition: ${extra}"
+          s"struct definition: ${keys1}, has members that do not appear in struct ${name}"
```
Comment (jdidion) [inline core/src/main/scala/dx/core/languages/wdl/WdlUtils.scala]: same problem here

File: core/src/main/scala/dx/core/languages/wdl/WdlUtils.scala [hunk 4]
```diff
@@ -602,10 +601,9 @@ object WdlUtils {
         // in memberTypes that do not appear in members are optional
         val keys1 = members.keySet
         val keys2 = memberTypes.keySet
-        val extra = keys2.diff(keys1)
-        if (extra.nonEmpty) {
+        if (!keys1.subsetOf(keys2)) {
           throw new Exception(
-              s"struct ${structName} value has members that do not appear in the struct definition: ${extra}"
+              s"struct definition: ${keys1}, has members that do not appear in struct ${structName}"
```
Comment (jdidion) [inline core/src/main/scala/dx/core/languages/wdl/WdlUtils.scala]: same problem here

**Review by jdidion — CHANGES_REQUESTED** [review]

File: core/src/main/scala/dx/core/ir/ValueSerde.scala (hunk 1 above)
Comment (mhrvol) [reply core/src/main/scala/dx/core/ir/ValueSerde.scala]: Check if the current error message is correct, feel free to update it if needed.

**Review by jdidion — DISMISSED** [review]

**Comment by jdidion** [comment]
Approved, but please don't merge until tests pass

**Review by jdidion — APPROVED** [review]

**Pull request merged**